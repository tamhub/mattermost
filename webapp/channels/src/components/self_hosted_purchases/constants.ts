// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const STORAGE_KEY_PURCHASE_IN_PROGRESS = 'PURCHASE_IN_PROGRESS';
export const STORAGE_KEY_EXPANSION_IN_PROGRESS = 'EXPANSION_IN_PROGRESS';
