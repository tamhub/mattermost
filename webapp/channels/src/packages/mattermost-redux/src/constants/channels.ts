// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const NotificationLevel = {
    DEFAULT: 'default',
    ALL: 'all',
    MENTION: 'mention',
    NONE: 'none',
};

export const MarkUnread = {
    ALL: 'all',
    MENTION: 'mention',
};
