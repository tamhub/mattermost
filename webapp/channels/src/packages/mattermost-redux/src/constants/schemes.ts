// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export const ScopeTypes = {
    TEAM: 'team',
    CHANNEL: 'channel',
};
