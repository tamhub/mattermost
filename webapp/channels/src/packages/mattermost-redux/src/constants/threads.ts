// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    THREADS_CHUNK_SIZE: 20,
};
