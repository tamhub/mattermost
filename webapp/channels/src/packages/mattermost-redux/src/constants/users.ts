// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

export default {
    IGNORE_CHANNEL_MENTIONS_ON: 'on',
    IGNORE_CHANNEL_MENTIONS_OFF: 'off',
    IGNORE_CHANNEL_MENTIONS_DEFAULT: 'default',
};
