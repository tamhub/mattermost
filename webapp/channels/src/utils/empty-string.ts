// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

// used as a stub for tsc type checker
export default '';
