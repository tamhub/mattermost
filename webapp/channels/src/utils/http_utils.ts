// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {useLocation} from 'react-router-dom';

export function useQuery() {
    return new URLSearchParams(useLocation().search);
}
