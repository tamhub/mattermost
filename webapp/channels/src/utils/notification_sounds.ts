// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import bing from 'sounds/bing.mp3';
import calls_calm from 'sounds/calls_calm.mp3';
import calls_cheerful from 'sounds/calls_cheerful.mp3';
import calls_dynamic from 'sounds/calls_dynamic.mp3';
import calls_urgent from 'sounds/calls_urgent.mp3';
import crackle from 'sounds/crackle.mp3';
import down from 'sounds/down.mp3';
import hello from 'sounds/hello.mp3';
import ripple from 'sounds/ripple.mp3';
import upstairs from 'sounds/upstairs.mp3';
import * as UserAgent from 'utils/user_agent';

export const notificationSounds = new Map([
    ['Bing', bing],
    ['Crackle', crackle],
    ['Down', down],
    ['Hello', hello],
    ['Ripple', ripple],
    ['Upstairs', upstairs],
]);

export const callsNotificationSounds = new Map([
    ['Dynamic', calls_dynamic],
    ['Calm', calls_calm],
    ['Urgent', calls_urgent],
    ['Cheerful', calls_cheerful],
]);

let canDing = true;
export function ding(name: string) {
    if (hasSoundOptions() && canDing) {
        tryNotificationSound(name);
        canDing = false;
        setTimeout(() => {
            canDing = true;
        }, 3000);
    }
}

export function tryNotificationSound(name: string) {
    const audio = new Audio(notificationSounds.get(name) ?? notificationSounds.get('Bing'));
    audio.play();
}

let currentRing: HTMLAudioElement | null = null;
export function ring(name: string) {
    if (!hasSoundOptions()) {
        return;
    }
    stopRing();

    currentRing = loopNotificationRing(name);
    currentRing.addEventListener('pause', () => {
        stopRing();
    });
}

export function stopRing() {
    if (currentRing) {
        currentRing.pause();
        currentRing.src = '';
        currentRing.remove();
        currentRing = null;
    }
}

let currentTryRing: HTMLAudioElement | null = null;
let currentTimer: NodeJS.Timeout;
export function tryNotificationRing(name: string) {
    if (!hasSoundOptions()) {
        return;
    }
    stopTryNotificationRing();
    clearTimeout(currentTimer);

    currentTryRing = loopNotificationRing(name);
    currentTryRing.addEventListener('pause', () => {
        stopTryNotificationRing();
    });

    currentTimer = setTimeout(() => {
        stopTryNotificationRing();
    }, 5000);
}

export function stopTryNotificationRing() {
    if (currentTryRing) {
        currentTryRing.pause();
        currentTryRing.src = '';
        currentTryRing.remove();
        currentTryRing = null;
    }
}

export function loopNotificationRing(name: string) {
    const audio = new Audio(callsNotificationSounds.get(name) ?? callsNotificationSounds.get('Calm'));
    audio.loop = true;
    audio.play();
    return audio;
}

export function hasSoundOptions() {
    return (!UserAgent.isEdge());
}
