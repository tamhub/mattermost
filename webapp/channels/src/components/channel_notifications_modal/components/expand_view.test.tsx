// Copyright (c) 2015-present Tam Dev, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import ExpandView from 'components/channel_notifications_modal/components/expand_view';

import {NotificationLevels, NotificationSections} from 'utils/constants';

jest.mock('react-redux', () => ({
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(() => true),
}));

describe('components/channel_notifications_modal/ExpandView', () => {
    const baseProps = {
        section: NotificationSections.DESKTOP,
        memberNotifyLevel: NotificationLevels.ALL,
        memberThreadsNotifyLevel: NotificationLevels.ALL,
        globalNotifyLevel: NotificationLevels.DEFAULT,
        serverError: '',
        onChange: jest.fn(),
        onChangeThreads: jest.fn(),
        onCollapseSection: jest.fn(),
        onSubmit: jest.fn(),
        onReset: jest.fn(),
        isGM: false,
    };

    describe('normal channels', () => {
        test('should match snapshot, DESKTOP on expanded view', () => {
            const wrapper = shallow(
                <ExpandView {...baseProps}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should match snapshot, PUSH on expanded view', () => {
            const props = {...baseProps, section: NotificationSections.PUSH};
            const wrapper = shallow(
                <ExpandView {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should match snapshot, MARK_UNREAD on expanded view', () => {
            const props = {...baseProps, section: NotificationSections.MARK_UNREAD};
            const wrapper = shallow(
                <ExpandView {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should match snapshot, DESKTOP on expanded view when mentions is selected', () => {
            const props = {...baseProps, memberNotifyLevel: NotificationLevels.MENTION};
            const wrapper = shallow(
                <ExpandView {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should match snapshot, PUSH on expanded view when mentions is selected', () => {
            const props = {...baseProps, section: NotificationSections.PUSH, memberNotifyLevel: NotificationLevels.MENTION};
            const wrapper = shallow(
                <ExpandView {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('gms', () => {
        test('should match snapshot, DESKTOP on expanded view when mentions is selected', () => {
            const props = {...baseProps, isGM: true, memberNotifyLevel: NotificationLevels.MENTION};
            const wrapper = shallow(
                <ExpandView {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });

        test('should match snapshot, PUSH on expanded view when mentions is selected', () => {
            const props = {...baseProps, section: NotificationSections.PUSH, isGM: true, memberNotifyLevel: NotificationLevels.MENTION};
            const wrapper = shallow(
                <ExpandView {...props}/>,
            );

            expect(wrapper).toMatchSnapshot();
        });
    });
});
