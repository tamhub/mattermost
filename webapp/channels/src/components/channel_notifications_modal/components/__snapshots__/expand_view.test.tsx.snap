// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/channel_notifications_modal/ExpandView gms should match snapshot, DESKTOP on expanded view when mentions is selected 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send desktop notifications"
              id="channel_notifications.sendDesktop"
            />
          </legend>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={false}
                id="channelNotificationAllActivity"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="all"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="all"
                section="desktop"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={true}
                id="channelNotificationMentions"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="mention"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="mention"
                section="desktop"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                id="channelNotificationNever"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="none"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="none"
                section="desktop"
              />
            </label>
          </div>
        </fieldset>
        <div
          className="mt-5"
        >
          <ExtraInfo
            section="desktop"
          />
        </div>
        <React.Fragment>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Notification sound"
                id="channel_notifications.sound"
              />
            </legend>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelDesktopSoundOn"
                  name="channelDesktopSound"
                  type="radio"
                  value="on"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="On"
                  id="channel_notifications.sound.on.title"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelDesktopSoundOff"
                  name="channelDesktopSound"
                  type="radio"
                  value="off"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="Off"
                  id="channel_notifications.sound.off.title"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Notification sounds are available on Firefox, Edge, Safari, Chrome and Tam Dev Desktop Apps."
                id="channel_notifications.sound_info"
              />
            </div>
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <SectionTitle
      isExpanded={true}
      onClickResetButton={[MockFunction]}
      section="desktop"
    />
  }
  updateSection={[MockFunction]}
/>
`;

exports[`components/channel_notifications_modal/ExpandView gms should match snapshot, PUSH on expanded view when mentions is selected 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send mobile push notifications"
              id="channel_notifications.sendMobilePush"
            />
          </legend>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={false}
                id="channelNotificationAllActivity"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="all"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="all"
                section="push"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={true}
                id="channelNotificationMentions"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="mention"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="mention"
                section="push"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                id="channelNotificationNever"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="none"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="none"
                section="push"
              />
            </label>
          </div>
        </fieldset>
        <div
          className="mt-5"
        >
          <ExtraInfo
            section="push"
          />
        </div>
      </div>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <SectionTitle
      isExpanded={true}
      onClickResetButton={[MockFunction]}
      section="push"
    />
  }
  updateSection={[MockFunction]}
/>
`;

exports[`components/channel_notifications_modal/ExpandView normal channels should match snapshot, DESKTOP on expanded view 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send desktop notifications"
              id="channel_notifications.sendDesktop"
            />
          </legend>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={true}
                id="channelNotificationAllActivity"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="all"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="all"
                section="desktop"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={false}
                id="channelNotificationMentions"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="mention"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="mention"
                section="desktop"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                id="channelNotificationNever"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="none"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="none"
                section="desktop"
              />
            </label>
          </div>
        </fieldset>
        <div
          className="mt-5"
        >
          <ExtraInfo
            section="desktop"
          />
        </div>
        <React.Fragment>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Notification sound"
                id="channel_notifications.sound"
              />
            </legend>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelDesktopSoundOn"
                  name="channelDesktopSound"
                  type="radio"
                  value="on"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="On"
                  id="channel_notifications.sound.on.title"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelDesktopSoundOff"
                  name="channelDesktopSound"
                  type="radio"
                  value="off"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="Off"
                  id="channel_notifications.sound.off.title"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Notification sounds are available on Firefox, Edge, Safari, Chrome and Tam Dev Desktop Apps."
                id="channel_notifications.sound_info"
              />
            </div>
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <SectionTitle
      isExpanded={true}
      onClickResetButton={[MockFunction]}
      section="desktop"
    />
  }
  updateSection={[MockFunction]}
/>
`;

exports[`components/channel_notifications_modal/ExpandView normal channels should match snapshot, DESKTOP on expanded view when mentions is selected 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send desktop notifications"
              id="channel_notifications.sendDesktop"
            />
          </legend>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={false}
                id="channelNotificationAllActivity"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="all"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="all"
                section="desktop"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={true}
                id="channelNotificationMentions"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="mention"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="mention"
                section="desktop"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                id="channelNotificationNever"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="none"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="none"
                section="desktop"
              />
            </label>
          </div>
        </fieldset>
        <div
          className="mt-5"
        >
          <ExtraInfo
            section="desktop"
          />
        </div>
        <React.Fragment>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Thread reply notifications"
                id="user.settings.notifications.threads.desktop"
              />
            </legend>
            <div
              className="checkbox"
            >
              <label>
                <input
                  checked={true}
                  id="desktopThreadsNotificationAllActivity"
                  name="desktopThreadsNotificationLevel"
                  onChange={[MockFunction]}
                  type="checkbox"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="Notify me about threads I'm following"
                  id="user.settings.notifications.threads.allActivity"
                />
              </label>
              <br />
            </div>
            <div
              className="mt-5"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="When enabled, any reply to a thread you're following will send a desktop notification."
                id="user.settings.notifications.threads"
              />
            </div>
          </fieldset>
        </React.Fragment>
        <React.Fragment>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Notification sound"
                id="channel_notifications.sound"
              />
            </legend>
            <div
              className="radio"
            >
              <label
                className=""
              >
                <input
                  checked={false}
                  id="channelDesktopSoundOn"
                  name="channelDesktopSound"
                  type="radio"
                  value="on"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="On"
                  id="channel_notifications.sound.on.title"
                />
              </label>
            </div>
            <div
              className="radio"
            >
              <label>
                <input
                  checked={false}
                  id="channelDesktopSoundOff"
                  name="channelDesktopSound"
                  type="radio"
                  value="off"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="Off"
                  id="channel_notifications.sound.off.title"
                />
              </label>
            </div>
            <div
              className="mt-5"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Notification sounds are available on Firefox, Edge, Safari, Chrome and Tam Dev Desktop Apps."
                id="channel_notifications.sound_info"
              />
            </div>
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <SectionTitle
      isExpanded={true}
      onClickResetButton={[MockFunction]}
      section="desktop"
    />
  }
  updateSection={[MockFunction]}
/>
`;

exports[`components/channel_notifications_modal/ExpandView normal channels should match snapshot, MARK_UNREAD on expanded view 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <fieldset>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={false}
                id="channelNotificationUnmute"
                name="channelNotificationMute"
                onChange={[MockFunction]}
                type="radio"
                value="mention"
              />
              <Describe
                memberNotifyLevel="mention"
                section="markUnread"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={true}
                id="channelNotificationMute"
                name="channelNotificationMute"
                onChange={[MockFunction]}
                type="radio"
                value="all"
              />
              <Describe
                memberNotifyLevel="all"
                section="markUnread"
              />
            </label>
          </div>
        </fieldset>
        <div
          className="mt-5"
        >
          <ExtraInfo
            section="markUnread"
          />
        </div>
      </div>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <SectionTitle
      isExpanded={true}
      onClickResetButton={[MockFunction]}
      section="markUnread"
    />
  }
  updateSection={[MockFunction]}
/>
`;

exports[`components/channel_notifications_modal/ExpandView normal channels should match snapshot, PUSH on expanded view 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send mobile push notifications"
              id="channel_notifications.sendMobilePush"
            />
          </legend>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={true}
                id="channelNotificationAllActivity"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="all"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="all"
                section="push"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={false}
                id="channelNotificationMentions"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="mention"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="mention"
                section="push"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                id="channelNotificationNever"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="none"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="none"
                section="push"
              />
            </label>
          </div>
        </fieldset>
        <div
          className="mt-5"
        >
          <ExtraInfo
            section="push"
          />
        </div>
      </div>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <SectionTitle
      isExpanded={true}
      onClickResetButton={[MockFunction]}
      section="push"
    />
  }
  updateSection={[MockFunction]}
/>
`;

exports[`components/channel_notifications_modal/ExpandView normal channels should match snapshot, PUSH on expanded view when mentions is selected 1`] = `
<SettingItemMax
  containerStyle=""
  infoPosition="bottom"
  inputs={
    Array [
      <div>
        <fieldset>
          <legend
            className="form-legend"
          >
            <Memo(MemoizedFormattedMessage)
              defaultMessage="Send mobile push notifications"
              id="channel_notifications.sendMobilePush"
            />
          </legend>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={false}
                id="channelNotificationAllActivity"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="all"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="all"
                section="push"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label
              className=""
            >
              <input
                checked={true}
                id="channelNotificationMentions"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="mention"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="mention"
                section="push"
              />
            </label>
          </div>
          <div
            className="radio"
          >
            <label>
              <input
                checked={false}
                id="channelNotificationNever"
                name="channelNotifications"
                onChange={[MockFunction]}
                type="radio"
                value="none"
              />
              <Describe
                globalNotifyLevel="default"
                memberNotifyLevel="none"
                section="push"
              />
            </label>
          </div>
        </fieldset>
        <div
          className="mt-5"
        >
          <ExtraInfo
            section="push"
          />
        </div>
        <React.Fragment>
          <hr />
          <fieldset>
            <legend
              className="form-legend"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="Thread reply notifications"
                id="user.settings.notifications.threads.push"
              />
            </legend>
            <div
              className="checkbox"
            >
              <label>
                <input
                  checked={true}
                  id="pushThreadsNotificationAllActivity"
                  name="pushThreadsNotificationLevel"
                  onChange={[MockFunction]}
                  type="checkbox"
                />
                <Memo(MemoizedFormattedMessage)
                  defaultMessage="Notify me about threads I'm following"
                  id="user.settings.notifications.push_threads.allActivity"
                />
              </label>
              <br />
            </div>
            <div
              className="mt-5"
            >
              <Memo(MemoizedFormattedMessage)
                defaultMessage="When enabled, any reply to a thread you're following will send a mobile push notification."
                id="user.settings.notifications.push_threads"
              />
            </div>
          </fieldset>
        </React.Fragment>
      </div>,
    ]
  }
  saving={false}
  section=""
  serverError=""
  submit={[MockFunction]}
  title={
    <SectionTitle
      isExpanded={true}
      onClickResetButton={[MockFunction]}
      section="push"
    />
  }
  updateSection={[MockFunction]}
/>
`;
